docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,692 @@
+
+Here's an example of how to configure SocketsHttpHandler with various properties and use it with HttpClient:
+
+```c#
+var cookieContainer = new CookieContainer();
+cookieContainer.Add(new Cookie("cookieName", "cookieValue"));
+
+var handler = new SocketsHttpHandler
+{
+    AllowAutoRedirect = true,
+    AutomaticDecompression = DecompressionMethods.All,
+    Expect100ContinueTimeout = TimeSpan.FromSeconds(1),
+    CookieContainer = cookieContainer,
+    Credentials = new NetworkCredential("user", "pass"),
+    MaxAutomaticRedirections = 10,
+    MaxResponseHeadersLength = 1,
+    Proxy = new WebProxy("http://proxyserver:8080"), // Don't forget to set UseProxy
+    UseProxy = true,
+    SslOptions = new SslClientAuthenticationOptions
+    {
+        RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
+        {
+            // Custom validation logic
+            return sslPolicyErrors == SslPolicyErrors.None;
+        }
+    },
+};
+
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+```
+
+### Example: Change ImpersonationLevel
+
+> [!WARNING]
+> This example contains the use of reflection, which may affect the performance of your application and can be broken in future versions of .NET. Use reflection with caution and only when necessary.
+
+In this example, we will use reflection to set the ImpersonationLevel property in SocketsHttpHandler:
+
+```csharp
+var handler = new SocketsHttpHandler();
+var settings = typeof(SocketsHttpHandler).GetField("_settings", BindingFlags.NonPublic | BindingFlags.Instance)?.GetValue(handler);
+Debug.Assert(settings != null);
+FieldInfo? fi = Type.GetType("System.Net.Http.HttpConnectionSettings, System.Net.Http")?.GetField("_impersonationLevel", BindingFlags.NonPublic | BindingFlags.Instance);
+fi?.SetValue(settings, TokenImpersonationLevel.Impersonation);
+
+var client = new HttpClient(handler);
+var response = await client.GetAsync(uri);
+```
+
+## Usage of Certificate and TLS Related Properties in HttpClient
+
+When working with `HttpClient`, you may need to handle client certificates for various purposes, such as custom validation of server certificates or fetching the server certificate. `HttpClient` provides several properties and options to manage certificates effectively.
+
+### Example: Enabling CRL Check with SocketsHttpHandler
+
+The `CheckCertificateRevocationList` property in `SocketsHttpHandler` allows developers to enable or disable the check for certificate revocation lists (CRL) during SSL/TLS handshake. Enabling this property ensures that the client verifies whether the server's certificate has been revoked, enhancing the security of the connection.
+
+**Old Code Using `HttpWebRequest`**:
+
+```csharp
+ServicePointManager.CheckCertificateRevocationList = true;
+HttpWebRequest request = (HttpWebRequest)WebRequest.Create(uri);
```

Comment:
Nit: some places use `CreateHttp` instead of `Create`. Might as well be consistent, unless this was intentional?